[JAVA] Improve map/flatmap of Publisher[Void] (#499)
mongodb/docs-java: source/whats-new.txt

  ``Decimal128`` values. In particular, the error responses for the
  ``isNumber()`` and ``asNumber()`` methods match the Java responses for
  equivalent ``Decimal128`` values.
  

> - This driver deprecates the ``org.mongodb.scala.Observable.completeWithUnit`` method

Reviewer: S: For consistency with other bullets and to avoid repeating "this driver," which is already in the intro

```suggestion
- The ``org.mongodb.scala.Observable.completeWithUnit()`` method is deprecated.
```

> - This driver deprecates the ``org.mongodb.scala.Observable.completeWithUnit`` method

Reviewer: S: Could be helpful to include what, if anything, users should use instead

Author: This is related to the breaking change about Observables - I've added a reference to the breaking change description here.
